### PR TITLE
Fix crash on OS X node tests

### LINKED ISSUE #8812: [node] Crash when running stress tests
I let the `memory.test.js` running overnight, trying to render 1200000 times with 10 maps on the pool. It blew up with a SEGV.

```
(master) tmpsantos@macbook:~/Projects/mapbox-gl-native$ BUILDTYPE=Release make node -j8 && time node --expose-gc platform/node/test/memory.test.js 
platform/linux/ninja  -j4 -C build/linux-x86_64/Release mbgl-node
ninja: Entering directory `build/linux-x86_64/Release'
ninja: no work to do.
TAP version 13
# Memory
# Rendering (0/1200000)
# Rendering (120000/1200000)
# Rendering (240000/1200000)
# Rendering (360000/1200000)
# Rendering (480000/1200000)
# Rendering (600000/1200000)
# Rendering (720000/1200000)
# Rendering (840000/1200000)
# Rendering (960000/1200000)
# Rendering (1080000/1200000)
Segmentation fault (core dumped)

real	458m20.525s
user	1088m48.156s
sys	51m18.880s
```

The `dmesg` was showing:

```
[1858468.176996] traps: node[1284] general protection ip:7fd906370d43 sp:7fff789a8ee8 error:0 in mapbox_gl_native.node[7fd90623b000+3c6000]
```

With this backtrace:

```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007fd906370d43 in std::_Function_handler<void (), node_mbgl::NodeRequest::Execute()::{lambda()#1}>::_M_invoke(std::_Any_data const&)
    () from /home/tmpsantos/Projects/mapbox-gl-native/lib/mapbox_gl_native.node
[Current thread is 1 (Thread 0x7fd90ca17740 (LWP 1284))]
(gdb) bt
#0  0x00007fd906370d43 in std::_Function_handler<void (), node_mbgl::NodeRequest::Execute()::{lambda()#1}>::_M_invoke(std::_Any_data const&)
    () from /home/tmpsantos/Projects/mapbox-gl-native/lib/mapbox_gl_native.node
#1  0x00007fd90c404703 in ?? () from /usr/lib/x86_64-linux-gnu/libuv.so.1
#2  0x00007fd90c4047e6 in ?? () from /usr/lib/x86_64-linux-gnu/libuv.so.1
#3  0x00007fd90c4136d8 in ?? () from /usr/lib/x86_64-linux-gnu/libuv.so.1
#4  0x00007fd90c4050ac in uv_run () from /usr/lib/x86_64-linux-gnu/libuv.so.1
#5  0x000055a499562338 in node::StartNodeInstance (arg=<synthetic pointer>) at ../src/node.cc:4084
#6  node::Start (argc=2, argv=<optimized out>) at ../src/node.cc:4160
#7  0x00007fd90a2b33f1 in __libc_start_main (main=0x55a4990a4270 <main(int, char**)>, argc=3, argv=0x7fff789ac838, init=<optimized out>, 
    fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fff789ac828) at ../csu/libc-start.c:291
#8  0x000055a4990a45a9 in _start ()
```

I'm going to run it again printing metrics about the V8 heap growth and see if this could be related.

/cc @bsudekum

### PR DESCRIPTION
**Pull request by jfirebaugh**
Fixes #8812.

### CODE REVIEW

File: platform/node/index.js [hunk 1]
```diff
@@ -19,7 +19,12 @@ var Map = function(options) {
     return new constructor(Object.assign(options, {
         request: function(req) {
             request(req, function() {
-                req.respond.apply(req, arguments);
+                var args = arguments;
+                // Protect ourselves from `request` implementations that try to release Zalgo.
```
Comment (kkaefer) [inline platform/node/index.js]: Not sure what *Zalgo* is without googling. Can we use a clearer phrasing of the issue here please?

File: platform/node/index.js [hunk 2]
```diff
@@ -19,7 +19,12 @@ var Map = function(options) {
...
+                // http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony
+                setImmediate(function() {
```
Comment (kkaefer) [inline platform/node/index.js]: Any reason to use `setImmediate` instead of `process.nextTick`?

File: platform/node/index.js (hunk 1 above)
Comment (jfirebaugh) [inline platform/node/index.js]: Did you see the link on the subsequent line?

File: platform/node/index.js (hunk 2 above)
Comment (jfirebaugh) [inline platform/node/index.js]: Not really. Sounds like `process.nextTick` is in fact more "immediate" than `setImmediate`. It probably doesn't matter though.

File: platform/node/index.js (hunk 1 above)
Comment (jfirebaugh) [inline platform/node/index.js]: Sorry, that was overly snarky. I've rephrased the comment, and kept the link for further background.

Comment (kkaefer) [inline platform/node/index.js]: Yeah, I saw the link, but even then it took me a while to understand that it refers to a meme. Thanks for updating!

**Pull request merged**